packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -588,26 +610,247 @@ export class UiControlClient extends ApiCommands {
       }
     },
   ) {
-    let command = this.click().text();
+    let command: FluentFiltersOrRelations | FluentFiltersOrRelationsGetter = this.click().text();
+
+    command = await this.evaluateMatchingProperty(command, params.type, params.text);
+
+    if (params.relation) {
+      command = this.evaluateRelation(command, params.relation.type, params.relation.text);
+    }
+
+    await command.exec();
+  }
 
-    switch (params.type) {
+  // eslint-disable-next-line class-methods-use-this
+  private async evaluateFinalExpectExistenceCommand(
+    finalCommand: FluentFiltersOrRelationsGetter,
+    element: ExpectExistenceInputParameter,
+  ): Promise<ExpectExistenceInputParameter> {
+    const el = element;
+    el.exists = false;
+    if ((await finalCommand.exec()).length > 0) {
+      el.exists = true;
+    }
+    return new Promise<ExpectExistenceInputParameter>((resolve) => {
+      resolve(el);
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private async evaluateMatchingProperty(
```

Comment:
I think this does not need to be an async function. At least I don't see why. You can directly return from the cases of the switch statement. That would make the method way less complex.